Fix decompress problem

VKlayd opened the pull request:
There was a problem when decompressed data were more than twice bigger
than compressed one.
On memory expanding by setlen something went wrong. As a result we got
lots of '\0' in the result data beginning.
Probably it'd happened because setlen does not copy data to reallocated
memory.
I fixed it by replacing setlen to simple string concatenation.

Araq commented:
But setLen should copy the data over after a reallocation!

VKlayd commented:
Should but dont.

I test this small code:

```
var s = newStringOfCap(50)
var teststr = "dfgh"
copyMem(addr(s[0]),addr(teststr[0]),4)
echo s
s.setlen(100)
echo s
```

very similar with code in zip module.
And got

```
dfgh

```

My fix is about a quickfix for problem and probably not good idea. =)
I novice in Nim and not sure how string copied to new place in `resizeString` (sysstr.nim). Time ago there was copyMem, but now it's commented.

Araq commented:
The code is wrong, this code works:

``` nim
var s = newStringOfCap(50)
var teststr = "dfgh"
copyMem(addr(s[0]),addr(teststr[0]),4)
setLen(s, 4)
echo s
s.setlen(100)
echo s
```

Note that after the first copyMem you need to set the size because it's used to determine what to do in a reallocation.

Varriount commented:
Similarly, the zlib code uses newStringOfCap, but only calls setlen when the string actually needs to be resized.

VKlayd committed:
Fix decompression problem

There was a problem when decompressed data were more than twice bigger
than compressed one
On memory expanding by setlen something went wrong. As a result we got
lots of '\0' in the result data beginning.
It happened because setLen not used before expand memory.

VKlayd commented:
Varriount, exactly. I fixed merge request. Now setLen first time calls with old space and second time with expanded value.

VKlayd commented:
Somebody wants to confirm this change or make any suggestions?

dom96 commented:
I would love to but I'm not familiar with this code. @Araq?
